### PR TITLE
Docs 4713 april20

### PR DESCRIPTION
**Pull request by valkyrie69**

### CODE REVIEW

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 1]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
...
+----
+PUT /oauth2/connect/register/{{client_id}}
+Accept: application/json
+Host: https://identity.example.com
+Authorization: Bearer access-token
+ {
+     "client_id": "client_id",
+     "client_secret": "some-secret",
+     "redirect_uris": [
+          "test-redirect-uri"
```
Comment (sebastiankorol) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]: Can we make this an URI? e.g. https://example.com/

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 2]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
...
+      ],
+     "grant_types": [ ],
```
Comment (sebastiankorol) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]:
```suggestion
     "grant_types": [ "authorization_code" ],
```

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 3]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
  In *Advanced Settings* you can also select:
   *** *Disable server certificate validation*: Disables server certificate validation if your OpenID client management instance presents a self-signed certificate, or one signed by an internal certificate authority.
   *** *Enable client deletion in Anypoint Platform*: Enables deletion of clients created with this integration.
-  *** *Enable client deletion and updates in IdP*: To use this option, you must also select the *Enable client deletion in Anypoint Platform* option. This option enables you to update and delete external clients in the configured IdP through an outbound call made by Anypoint Platform. 
+  *** *Enable client deletion and updates in IdP*: To use this option, you must also select the *Enable client deletion in Anypoint Platform* option. This option enables you to update and delete external clients in the configured IdP through an outbound call made by Anypoint Platform to `{clientRegistrationUrl}/{clientID}`, for example: `+https://example.salesforce.com/services/oauth2/6779ef20e75817b79602+`.
```
Comment (santro92) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]: It will also be good to make it clear that `https://example.salesforce.com/services/oauth2/` is the client registration URL they configured and `6779ef20e75817b79602` is the clientId

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 4]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
...
+Authorization: Bearer access-token
+ {
+     "client_id": "client_id",
+     "client_secret": "some-secret",
+     "redirect_uris": [
+          "test-redirect-uri"
+      ],
+     "grant_types": [ ],
+     "token_endpoint_auth_method": "client_secret_basic",
+     "response_types": [ ],
```
Comment (sebastiankorol) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]:
```suggestion
     "response_types": [ "code" ],
```

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 5]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
...
+   **** The `Authorization:` header is included as part of the request only if the `Authorization Header` under *Advanced settings* is configured.
+   **** The `client_id` request parameter that is passed is the same as the `client_id` passed in the `PUT` request payload.
+   **** The `token_endpoint_auth_method` passed in the payload is always `client_secret_basic`.
+   **** Only the `Client Registration URL` and `Authorization Header` come from what is configured in OIDC *Client management*. +
+This is an example of the payload for a `PUT` (update) request: 
++
+----
+PUT /oauth2/connect/register/{{client_id}}
+Accept: application/json
+Host: https://identity.example.com
```
Comment (santro92) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]:
```suggestion
Host: identity.example.com
```

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 6]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
...
+     "response_types": [ ],
+     "client_name": "test-client-name"
+}
+----
++
+This is an example of the `DELETE` request header: 
++
+----
+DELETE /oauth2/connect/register/{{client_id}}
+Host: https://identity.example.com
```
Comment (santro92) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]:
```suggestion
Host: identity.example.com
```

File: modules/ROOT/pages/configure-client-management-openid-task.adoc (hunk 4 above)
Comment (sebastiankorol) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]: Given that this isn't directly translated from the API, we should say how it gets generated based on the grant_types:

```
authorization_code: ['code'],
implicit: ['token', 'id_token']
```

**Comment by santro92** [comment]
And a note saying the outbound call might not work right out of the box for all the providers, they need to subscribe/intercept the call and tune the request according to their IdP convention of achieving the same

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 7]
```diff
@@ -47,7 +47,39 @@ The *External Identity* page appears.
...
+ {
+     "client_id": "client_id",
+     "client_secret": "some-secret",
+     "redirect_uris": [
+          "test-redirect-uri"
+      ],
+     "grant_types": [ ],
+     "token_endpoint_auth_method": "client_secret_basic",
+     "response_types": [ ],
+     "client_name": "test-client-name"
```
Comment (sebastiankorol) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc:73]: I think this is fairly obvious, but this field doesn't map cleanly to the field that you would be PATCHing on the inbound side: it's `name` there. I think we could add a note here.

**Comment by sebastiankorol** [comment]
The note is a very good point. I would make it stronger though: The call is expected not to work if used directly against an IDP. It's intended for interception by pointing to an application under the user's control, so that they can implement the correct deletion/update APIs for their particular provider

File: modules/ROOT/pages/configure-client-management-openid-task.adoc (hunk 3 above)
Comment (valkyrie69) [reply modules/ROOT/pages/configure-client-management-openid-task.adoc]: I think since it's an example of {clientRegistrationUrl}/{clientID}, it's clear that 6779ef20e75817b79602 is the client ID.

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 8]
```diff
@@ -47,7 +47,47 @@ The *External Identity* page appears.
...
   *** *Enable client deletion in Anypoint Platform*: Enables deletion of clients created with this integration.
-  *** *Enable client deletion and updates in IdP*: To use this option, you must also select the *Enable client deletion in Anypoint Platform* option. This option enables you to update and delete external clients in the configured IdP through an outbound call made by Anypoint Platform. 
+  *** *Enable client deletion and updates in IdP*: To use this option, you must also select the *Enable client deletion in Anypoint Platform* option. This option enables you to update and delete external clients in the configured IdP through an outbound call made by Anypoint Platform to `{clientRegistrationUrl}/{clientID}`. The `clientRegistrationUrl` is the value you configured in *Client Registration URL*. For example: `+https://identity.example.com/oauth2/connect/register/6779ef20e75817b79602+`. +
+  
+   **** The defined `Client Registration URL` is the UI through which both `PUT` and `DELETE` requests are made.
```
Comment (santro92) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]: This point is not needed as we kind of state the same in the paragraph above

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 9]
```diff
@@ -47,7 +47,47 @@ The *External Identity* page appears.
...
+  *** *Enable client deletion and updates in IdP*: To use this option, you must also select the *Enable client deletion in Anypoint Platform* option. This option enables you to update and delete external clients in the configured IdP through an outbound call made by Anypoint Platform to `{clientRegistrationUrl}/{clientID}`. The `clientRegistrationUrl` is the value you configured in *Client Registration URL*. For example: `+https://identity.example.com/oauth2/connect/register/6779ef20e75817b79602+`. +
+  
+   **** The defined `Client Registration URL` is the UI through which both `PUT` and `DELETE` requests are made.  
+   **** The `Authorization:` header is included as part of the request only if the `Authorization Header` under *Advanced settings* is configured.
+   **** The `client_id` request parameter that is passed is the same as the `client_id` passed in the `PUT` request payload.
+   **** The `token_endpoint_auth_method` passed in the payload is always `client_secret_basic`.
+   **** Only the `Client Registration URL` and `Authorization Header` come from what is configured in OIDC *Client management*. +
+This is an example of the payload for a `PUT` (update) request:
```
Comment (santro92) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc:57]: We need spacing before this line, so it looks neat in the UI

File: modules/ROOT/pages/configure-client-management-openid-task.adoc [hunk 10]
```diff
@@ -47,7 +47,47 @@ The *External Identity* page appears.
...
+This is an example of the `DELETE` request header: 
++
+----
+DELETE /oauth2/connect/register/{{client_id}}
+Host: identity.example.com
+Authorization: Bearer access-token
+----
++
+[IMPORTANT]
+The call is not expected to work if it used directly against an IDP. It's intended for interception by pointing to an application under the user's control, so they can implement the correct deletion and update APIs for their particular provider.
```
Comment (sebastiankorol) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]: We talk in the second person about the reader typically. Should we do the same here?

Comment (santro92) [inline modules/ROOT/pages/configure-client-management-openid-task.adoc]: Can you reword it to `Both PUT and DELETE call are no expected ...`

File: modules/ROOT/pages/configure-client-management-openid-task.adoc (hunk 9 above)
Comment (valkyrie69) [reply modules/ROOT/pages/configure-client-management-openid-task.adoc:57]: Formatting is limited and finicky and very easy to mess up. So I'm leaving it alone.

**Review by santro92 — APPROVED** [review]
LGTM

**Review by sebastiankorol — APPROVED** [review]
